tree/ntuple/v7/src/RField.cxx

```diff
@@ -659,10 +692,12 @@ ROOT::Experimental::RFieldBase::Create(const std::string &fieldName, const std::
    if (!result) {
       auto cl = TClass::GetClass(canonicalType.c_str());
       if (cl != nullptr) {
-         if (cl->GetCollectionProxy())
+         if (cl->GetCollectionProxy()) {
```

Comment:
The collection might contain a class, do where do we capture the cycle in this case?

Reply:
Yes, in the constructor of `RProxiedCollectionField`, we call `RFieldBase::Create` if we have an `GetValueClass()` for the collection proxy, and subsequently will detect the cycle.

Reply:
Oh, you mean: the collection proxy's value class is the collection proxy itself. Yes, I think you're right, we should move the adding the class to the context guard up.